File: Java/manasa-and-stones.java
class Result {
    public static List<Integer> stones(int n, int a, int b) {
        List<Integer> res = new ArrayList<>();
        if(a>b){
            int temp = a;
            a = b;
            b = temp;
        }
        int gap = b - a;
        int fav = a * (n-1);
        res.add(fav);
        if( gap == 0) return res;
        
        for(int i = 1;i<n;i++){
            res.add(fav + (gap * i));
        }
        return res;
    }
}

File: Python/manasa-and-stones.py
def stones(n, a, b):
   res = []
   if a > b:
     a , b = b , a
   gap = b - a
 
   fav = a * (n-1)
   res.append(fav)
 
   if gap == 0:
     return res
 
   for i in range(1 , n):
     res.append(fav + (gap*i))
   return res